packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -30,176 +30,237 @@ import { IL1ERC721Bridge } from "interfaces/L1/IL1ERC721Bridge.sol";
-        IFaultDisputeGame faultDisputeGame;
-        IPermissionedDisputeGame permissionedDisputeGame;
-        IDelayedWETH delayedWETHPermissionedGameProxy;
-        IDelayedWETH delayedWETHPermissionlessGameProxy;
+    function blueprints() public view returns (OPContractsManager.Blueprints memory) {
+        return blueprint;
     }
 
-    /// @notice Addresses of ERC-5202 Blueprint contracts. There are used for deploying full size
-    /// contracts, to reduce the code size of this factory contract. If it deployed full contracts
-    /// using the `new Proxy()` syntax, the code size would get large fast, since this contract would
-    /// contain the bytecode of every contract it deploys. Therefore we instead use Blueprints to
-    /// reduce the code size of this contract.
-    struct Blueprints {
-        address addressManager;
-        address proxy;
-        address proxyAdmin;
-        address l1ChugSplashProxy;
-        address resolvedDelegateProxy;
-        address permissionedDisputeGame1;
-        address permissionedDisputeGame2;
-        address permissionlessDisputeGame1;
-        address permissionlessDisputeGame2;
+    function implementations() public view returns (OPContractsManager.Implementations memory) {
+        return implementation;
     }
+}
 
-    /// @notice The latest implementation contracts for the OP Stack.
-    struct Implementations {
-        address superchainConfigImpl;
-        address protocolVersionsImpl;
-        address l1ERC721BridgeImpl;
-        address optimismPortalImpl;
-        address systemConfigImpl;
-        address optimismMintableERC20FactoryImpl;
-        address l1CrossDomainMessengerImpl;
-        address l1StandardBridgeImpl;
-        address disputeGameFactoryImpl;
-        address anchorStateRegistryImpl;
-        address delayedWETHImpl;
-        address mipsImpl;
+contract OPContractsManagerBase {
+    /// @notice The blueprint contract addresses contract.
+    OPContractsManagerContractsContainer public immutable contractsContainer;
+
+    /// @notice The OPContractsManager contract that is currently being used.
+    OPContractsManagerBase internal immutable thisOPCM;
+
+    /// @notice Constructor to initialize the immutable thisOPCM variable and contract addresses
+    /// @param _contractsContainer The blueprint contract addresses and implementation contract addresses
+    constructor(OPContractsManagerContractsContainer _contractsContainer) {
+        contractsContainer = _contractsContainer;
+        thisOPCM = this;
     }
 
-    /// @notice The input required to identify a chain for upgrading, along with new prestate hashes
-    struct OpChainConfig {
-        ISystemConfig systemConfigProxy;
-        IProxyAdmin proxyAdmin;
-        Claim absolutePrestate;
+    /// @notice Retrieves the implementation addresses stored in this OPCM contract
+    function getImplementations() internal view returns (OPContractsManager.Implementations memory) {
```

Comment:
Can we remove getImplementations() and getBlueprints() and all references to thisOPCM in this contract? Check